packages/taskserver/src/collect-analytics.ts

```diff
@@ -38,6 +38,9 @@ const DEFAULT_INTERVAL_SECONDS = 1800
 const configInterval = parseInt(config.taskserver.processInterval)
 const interval = (configInterval || DEFAULT_INTERVAL_SECONDS) * 1000
 
+let lastTimestamp: string
+let analyticsData: any[] = []
+const collectedData: any[] = []
```

Comment:
Will collectedData continue to grow in memory infinitely? I think it will also repeat a lot of data because of that. Unless you're filtering on the Opensource ingest side.